### PR TITLE
Update installation page to include istio-demo-auth story

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: content/docs/setup/kubernetes/quick-start.md [hunk 1]
```diff
@@ -268,11 +268,19 @@ customize Istio to your needs.
 
     _**OR**_
 
-    b)  [Render Kubernetes manifest with Helm and deploy with kubectl](/docs/setup/kubernetes/helm-install/#option-1-install-with-helm-via-helm-template).
+    b) Install Istio and enable mutual TLS authenticaiton globally between sidecars. This option is only recommended for new clusters, and all applications are installed with sidecars. Otherwise, we recommend to choose option a) and incrementally adopt Istio mutual TLS later on.
```
Comment (incfly) [inline content/docs/setup/kubernetes/quick-start.md]: > "mutual TLS authentication". 

When mTLS is turned on without authn policy, does authn happen? Or we should just say "mutual TLS" here. (also in previous paragraph).

Comment (incfly) [inline content/docs/setup/kubernetes/quick-start.md]: > incrementally adopt Istio mutual TLS later on.

Should we link to where @diemtvu 's doc about per-service enabling of mTLS by using DR and Authn Policy?

**Comment by incfly** [comment]
Nvm. Yeah I remember we infer the verify San based on mesh config flag
value .

**Comment by wattli** [comment]
@geeknoid , could you help to approve?

File: content/docs/setup/kubernetes/quick-start.md (hunk 1 above)
Comment (diemtvu) [inline content/docs/setup/kubernetes/quick-start.md]: authenticaiton -> authentication

Comment (diemtvu) [reply content/docs/setup/kubernetes/quick-start.md]: I think it's better to be a note under previous bullet point (a).

Comment (wattli) [reply content/docs/setup/kubernetes/quick-start.md]: Done.

File: content/docs/setup/kubernetes/quick-start.md [hunk 2]
```diff
@@ -268,11 +268,19 @@ customize Istio to your needs.
 
     _**OR**_
 
-    b)  [Render Kubernetes manifest with Helm and deploy with kubectl](/docs/setup/kubernetes/helm-install/#option-1-install-with-helm-via-helm-template).
+    b) Install Istio and enable mutual TLS authentication globally between sidecars. This option is only recommended for new clusters, and all applications are installed with sidecars. Otherwise, we recommend to choose option a) and incrementally adopt Istio mutual TLS later on.
+
```
Comment (rshriram) [inline content/docs/setup/kubernetes/quick-start.md]: grammar.
(c) Install Istio and enforce mutual TLS authentication by default. Use this option only on a fresh kubernetes cluster where newly deployed workloads are guaranteed to have Istio sidecars installed.

Comment (wattli) [reply content/docs/setup/kubernetes/quick-start.md]: done.

**Pull request merged**